zerver/views/development/integrations.py

```diff
@@ -0,0 +1,66 @@
+import os
+import ujson
+from typing import List
+
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.test import Client
+from django.utils.translation import ugettext as _
+
+from zerver.lib.integrations import WEBHOOK_INTEGRATIONS
+from zerver.lib.request import has_request_variables, REQ
+from zerver.lib.response import json_success, json_error
+from zerver.models import UserProfile, get_realm
+
+# TODO: Add support for headers. This will likely be accompanied by a change in the way we store our
+# fixtures, though that won't be required to support this feature.
+
+ZULIP_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../../../')
+
+
+def get_webhook_integrations() -> List[str]:
+    return [integration.name for integration in WEBHOOK_INTEGRATIONS]
+
+
+def dev_panel(request: HttpRequest) -> HttpResponse:
+    integrations = get_webhook_integrations()
+    bots = UserProfile.objects.filter(is_bot=True, bot_type=UserProfile.INCOMING_WEBHOOK_BOT)
+    context = {"integrations": integrations, "bots": bots}
+    return render(request, "zerver/integrations/development/dev_panel.html", context)
+
+
+@has_request_variables
+def check_send_webhook_fixture_message(request: HttpRequest,
+                                       url: str=REQ(),
+                                       body: str=REQ()) -> HttpResponse:
+    realm = get_realm("zulip")
+    client = Client()
+    response = client.post(url, body, content_type="application/json", HTTP_HOST=realm.host)
+    if response.status_code == 200:
+        return json_success({"msg": "Success"})
+    else:
+        return response
+
+
+@has_request_variables
+def get_fixtures(request: HttpResponse,
+                 integration_name: str=REQ()) -> HttpResponse:
+    integrations = get_webhook_integrations()
+    if integration_name not in integrations:
+        return json_error(_("\"%s\" is not a valid webhook integration.") % (integration_name),
+                          status=404)
+
+    fixtures_pack = {}
+    fixtures_dir = os.path.join(ZULIP_PATH, "zerver/webhooks/%s/fixtures" % (integration_name))
+    if not os.path.exists(fixtures_dir):
+        return json_error(_("The integration \"%s\" does not have fixtures."), status=404)
+    for fixture in os.listdir(fixtures_dir):
+        fixture_path = os.path.join(fixtures_dir, fixture)
+        try:
+            json_data = ujson.loads(open(fixture_path).read())
+        except ValueError:
+            return json_error(_("The integration \"%s\" does not have JSON fixtures.") % (
+                              integration_name))
```

Comment:
I would recommend using `"{arg}".format(arg=argval)` style formatting everywhere! The other style is old and we should try to use the new style wherever possible. It is more readable! :)

Reply:
Oh, sorry about that, I'll definitely make the change :+1:.